pipeline/pipeline.py

```diff
@@ -0,0 +1,58 @@
+from dataclasses import dataclass
+from typing import List, Optional
+
+from task import AbstractTask
+
+
+@dataclass
+class PipelineConfig:
+    """Configuration of the workspace migration pipeline."""
+    base_dir: str
+    session: str
+
+
+class Pipeline:
+    """Class that coordinates run of a group of tasks, which form a DAG based on dependencies
+    defined in add_task().
+
+    The pipeline has built-in checkpoint i.e it will skip all complete tasks upon restart."""
+
+    @dataclass
+    class Node:
+        task: AbstractTask = None
+        children = []
+
+    def __init__(self, checkpoint: Optional[str]):
+        self._source = self.Node()
+        self._checkpoint = checkpoint
+
+    def add_task(self, task: AbstractTask, parents: Optional[List[Node]] = None):
```

Comment:
According to the examples down below, it seems like this method should return the `node`?

Reply:
Yes, good catch!